src/models/document/document.ts

```diff
@@ -279,6 +287,18 @@ export const DocumentModel = Tree.named("Document")
 
     setProperties(properties: ISetProperties) {
       forEach(properties, (value, key) => self.setProperty(key, value));
+    },
+
+    setContentStatus(status: ContentStatus) {
+      self.contentStatus = status;
+    },
+
+    setInvalidContent(content: object) {
+      self.invalidContent = content;
+    },
+
+    setContentErrorMessage(message: string) {
+      self.contentErrorMessage = message;
```

Comment:
Seems like these would likely always be called together. Is there a reason to prefer three separate functions to a single function that sets all three, e.g. `setContentError()`? If we ever add the ability to fix a document we could add a `clearContentError()` function, but it seems like these are inherently related properties that could/should be set by a single action.

Reply:
Yeah I can consolidate those functions.